component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -543,4 +543,23 @@ public static String getEnableTOTPPageFromXMLFile(AuthenticationContext context,
 		}
 		return enableTOTPPage;
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Get the useEventHandlerBasedEmailSender config value from the application-authentication.xml file.
+	 *
+	 * @return Is Event Handler Based Email Sender Enabled.
+	 * @throws AuthenticationFailedException
+	 */
+	public static boolean isEventHandlerBasedEmailSenderEnabled() {
+
+		String eventHandlerBasedEmailSenderProperty = getTOTPParameters()
+				.get(TOTPAuthenticatorConstants.USE_EVENT_HANDLER_BASED_EMAIL_SENDER);
+		if (eventHandlerBasedEmailSenderProperty != null && Boolean
```

Comment:
You can directly, return Boolean.parseBoolean(eventHandlerBasedEmailSenderProperty);
As this returns false if the string is null or any other string. You do not need to do null check purposely.

Reply:
done